src/extension.ts

```
@@ -345,41 +357,39 @@ async function analyzeAllFiles(files: vscode.Uri[], token: vscode.CancellationTo
-                    resolve();
-                }
-            };
+                // Update current file index after each file is processed
+                currentFileIndex = i + 1;
+            }
             
-            processBatch(0);
+            // All files have been processed
+            createPDF(totalSmells);
+            resolve();
         });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)